### PR TITLE
feat: retry subscribe on failure

### PR DESCRIPTION
**Pull request by thalesmg**

### CODE REVIEW

File: src/emqtt_bench.erl [hunk 1]
```diff
@@ -458,6 +458,25 @@ inc_counter(connect_succ) ->
 inc_counter(connect_fail) ->
     counters:add(cnt_ref(), ?IDX_CONN_FAIL, 1).
 
+dec_counter(recv) ->
```
[the remark this answers is not in the document]
Comment (thalesmg) [reply src/emqtt_bench.erl:461]: I don't think so.  I just added all cases for consistency.

[the remark this answers is not in the document]
Comment (zmstone) [reply src/emqtt_bench.erl:461]: I’d recommend to only add possible clauses.
I.e. gauges, not counters

[the remark this answers is not in the document]
Comment (thalesmg) [reply src/emqtt_bench.erl]: I changed it to increase the success/failure counters only after retries are exhausted, so there's no need to fiddle with decreasing counters.

**Review by zmstone — APPROVED** [review]

File: src/emqtt_bench.erl [hunk 2]
```diff
@@ -515,35 +515,47 @@ connect(Parent, N, PubSub, Opts) ->
...
-                false ->
-                    io:format("client(~w): retrying...~n", [N]),
-                    NOpts = proplists:delete(connection_attempts, Opts),
-                    connect(Parent, N, PubSub, [{connection_attempts, Retries + 1} | NOpts])
-            end
+            maybe_retry(Parent, N, PubSub, Opts, ContinueFn)
+    end.
+
+maybe_retry(Parent, N, PubSub, Opts, ContinueFn) ->
+    MaxRetries = proplists:get_value(num_retry_connect, Opts, 0),
```
Comment (qzhuyan) [inline src/emqtt_bench.erl:548]: How do you set these?

Comment (qzhuyan) [reply src/emqtt_bench.erl:548]: oh sorry, I did not pull the latest master

**Review by qzhuyan — APPROVED** [review]

**Pull request merged**